src/MarketingPageAcceptanceTests.TestData/Suppliers/Supplier.cs

```diff
@@ -7,5 +11,37 @@ public sealed class Supplier
         public string LegalName { get; set; }
         public string Summary { get; set; }
         public string SupplierUrl { get; set; }
+        public DateTime LastUpdated { get; set; } = DateTime.Now;
+        public Guid LastUpdatedBy { get; set; } = Guid.Empty;
+
+        public void Create(string connectionString)
+        {
+            var query = Queries.CreateNewSupplier;
+            SqlExecutor.Execute<Supplier>(connectionString, query, this);
+        }
+
+        public Supplier Retrieve(string connectionString)
+        {
+            var query = Queries.GetSupplier;
+            return SqlExecutor.Execute<Supplier>(connectionString, query, this).Single();
+        }
+
+        public Supplier RetrieveSupplierForSolution(string connectionString, string solutionId)
```

Comment:
Can you use the generic Retrieve instead of this method? What would that look like in the steps?